fatiando/mesher.py

```diff
@@ -1721,3 +1721,594 @@ def keep(cell):
             return True
         return False
     return [c for c in cells if keep(c)]
+
+
+class TriaxialEllipsoid(GeometricElement):
+
+    """
+    Create an arbitrarily-oriented triaxial ellipsoid.
+
+    Triaxial ellipsoids are those having different semi-axes.
+    This code follows Clark et al. (1986) and defines the spatial
+    orientation of the ellipsoid by using three angles: strike,
+    rake and dip. These angles are commonly used to define the
+    orientation of geological structures (Allmendinger et al., 2012).
+
+    References:
+
+    Allmendinger, R., Cardozo, N., and Fisher, D. M.: Structural geology
+    algorithms : vectors and tensors, Cambridge University Press, 2012.
+
+    Clark, D., Saul, S., and Emerson, D.: Magnetic and gravity anomalies
+    of a triaxial ellipsoid, Exploration Geophysics, 17, 189-200, 1986.
+
+    .. note:: The coordinate system used is x -> North, y -> East
+    and z -> Down
+
+    Parameters:
+
+    * x, y, z : float
+        The coordinates of the center of the ellipsoid.
+    * large_axis, intermediate_axis, small_axis: float
+        Semi-axes forming the ellipsoid (in m).
+    * strike, dip, rake
+        Orientation angles of the ellipsoid (in degrees).
+    * props : dict
+        Physical properties assigned to the ellipsoid.
+        Ex: ``props={'density':10,
+                     'remanent magnetization':[10, 25, 40],
+                     'susceptibility tensor':[0.562, 0.485, 0.25,
+                                              90, 0, 0]}``
+
+    Examples:
+
+        >>> e = TriaxialEllipsoid(x=1, y=2, z=3, large_axis=6,
+        ...                       intermediate_axis=5, small_axis=4,
+        ...                       strike=10, dip=20, rake=30, props={
+        ...                       'remanent magnetization': [10, 25, 40],
+        ...                       'susceptibility tensor': [0.562, 0.485,
+        ...                                                 0.25, 90, 0,
+        ...                                                 0]})
+        >>> e.props['remanent magnetization']
+        [10, 25, 40]
+        >>> e.addprop('density', 20)
+        >>> print(e.props['density'])
+        20
+        >>> print(e)
+        x:1 | y:2 | z:3 | large_axis:6 | intermediate_axis:5 | small_axis:4 | \
+strike:10 | dip:20 | rake:30 | density:20 | remanent magnetization:[10, 25, 40\
+] | susceptibility tensor:[0.562, 0.485, 0.25, 90, 0, 0]
```

Comment:
> 1. string representation
> What do you think about breaking the string representation into lines?

I don't think that will help. It will still be ugly and not very useful. I prefer to just get rid of it. Take a look at what I've been doing in #378 (https://github.com/fatiando/fatiando/pull/378/files#diff-94a56fa30b5d3b4dc51db74b1bc2cae9). That PR ended up being too big and I'll break it into smaller parts later.

> 2. susceptibility tensor calculation
> The susceptibility tensor is a numpy array. The problem of creating a doctest illustrating the susceptibility tensor calculation is that it would be necessary to print a numpy array. What do you think?

Then we'd run into the precision problems we were having before. Hum. You're right. Maybe just leave the simple stuff in the doctests. This isn't a complicated class and there isn't much need for complex examples in the docstring. Use the test modules to test the tensor, as you've already been doing.